File: ToDoList/app/src/main/java/com/example/todolist/mmodel/CategoryModel.java
package com.example.todolist.mmodel;

import com.example.todolist.mclass.Category;

import java.util.ArrayList;

public class CategoryModel {
    private static ArrayList<Category> items;

    public static ArrayList<Category> getItems() {
        if (items == null) {
            items = new ArrayList<Category>();

            items.add(new Category("Category_Home", "Home", 1));
            items.add(new Category("Category_Working", "Work", 1));
            items.add(new Category("Category_Studying", "Study", 1));
            items.add(new Category("Category_Sport", "Sport", 1));
            items.add(new Category("Category_Entertainment", "Entertainment", 1));
            items.add(new Category("Category_Others", "Others", 1));

        }
        return items;
    }

    public static String getNameByID(String id) {
        if (items == null) {
            getItems();
        } else {
            for (int i = 0; i < items.size(); i++) {
                String getID = getItems().get(i).getId();
                if (getID.equalsIgnoreCase(id)) {
                    return getItems().get(i).getName();
                }
            }
        }
        return null;
    }

    public static Category getItemsByID(String id) {
        if (items == null) {
            getItems();
        } else {
            for (int i = 0; i < items.size(); i++) {
                String getID = getItems().get(i).getId();
                if (getID.equalsIgnoreCase(id)) {
                    return items.get(i);
                }
            }
        }
        return null;
    }
}

File: ToDoListKotlin/app/src/main/java/com/example/todolistkotlin/activity/model/CategoryModel.kt
package com.example.todolistkotlin.activity.model

import com.example.todolistkotlin.activity.classes.Category
import java.util.*


object CategoryModel {
    private var items: ArrayList<Category>? = null
    fun getItems(): ArrayList<Category>? {
        if (items == null) {
            items = ArrayList<Category>()
            items!!.add(Category("Category_Home", "Home", 1))
            items!!.add(Category("Category_Working", "Work", 1))
            items!!.add(Category("Category_Studying", "Study", 1))
            items!!.add(Category("Category_Sport", "Sport", 1))
            items!!.add(Category("Category_Entertainment", "Entertainment", 1))
            items!!.add(Category("Category_Others", "Others", 1))
        }
        return items
    }

    fun getNameByID(id: String?): String? {
        if (items == null) {
            getItems()
        } else {
            for (i in items!!.indices) {
                val getID: String = getItems()!![i].getId()
                if (getID.equals(id, ignoreCase = true)) {
                    return getItems()!![i].getName()
                }
            }
        }
        return null
    }

    fun getItemsByID(id: String?): Category? {
        if (items == null) {
            getItems()
        } else {
            for (i in items!!.indices) {
                val getID: String = getItems()!![i].getId()
                if (getID.equals(id, ignoreCase = true)) {
                    return items!![i]
                }
            }
        }
        return null
    }
}